### PR TITLE
Merge related repositories

### PR DESCRIPTION
**Pull request by sampsapenna**
### Description

The development of the sharing functionality, container access request functionality and upload+download enhancements is integrated enough with the main project, that it makes sense to develop them as one larger bundle. This PR moves the following repositories to the `swift-browser-ui` main repo:

* https://github.com/cscfi/swift-x-account-sharing
* https://github.com/cscfi/swift-sharing-request
* https://github.com/cscfi/swiftui-upload-runner

Additionally the PR removes instance specific customization and branding from the main repo, to be contained elsewhere.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (this needs a follow up PR)

### Changes Made

* Unify projects into a single repository
* Refactor python code into a single package
* Refactor tests to use the new package structure
* Remove workarounds in `package.json` and dockerfiles due to installed relevant APIs being a part of the main repo now
* Add separate launch commands for the additional services
* Add `swift-x-account-sharing`, `swift-sharing-request`, `swiftui-upload-runner`  to documentation module list
* Move to using absolute imports only
* Refactor dockerfiles and repository structure for building multiple services
* Remove customized dockerfiles and branding from the repository
* Add `swift-x-acount-sharing`, `swift-sharing-request`, `swiftui-upload-runner`  tests to the main tox testing pipeline
* Move `swift-x-account-sharing`, `swift-sharing-request` js bindings to frontend source, to fix docker issues with symlinks
* Remove dependency on `swift-x-account-sharing`, `swift-sharing-request` repositories with the repos being in the main repository

### Testing

- [x] Unit Tests
- [x] Integration Tests
- [ ] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: README-request.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+![Python style check](https://github.com/CSCfi/swift-sharing-request/workflows/Python%20style%20check/badge.svg)
+![Javascript ESLint check](https://github.com/CSCfi/swift-sharing-request/workflows/Javascript%20ESLint%20check/badge.svg)
+![Python Unit Tests](https://github.com/CSCfi/swift-sharing-request/workflows/Python%20Unit%20Tests/badge.svg)
```
Comment (blankdots) [inline README-request.md:3]: these might be forgotten things

File: README-sharing.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+![Javascript ESLint check](https://github.com/CSCfi/swift-x-account-sharing/workflows/Javascript%20ESLint%20check/badge.svg)
+![Python style check](https://github.com/CSCfi/swift-x-account-sharing/workflows/Python%20style%20check/badge.svg)
+![Python Unit Tests](https://github.com/CSCfi/swift-x-account-sharing/workflows/Python%20Unit%20Tests/badge.svg)
```
Comment (blankdots) [inline README-sharing.md:3]: these might be forgotten things

File: README.md [hunk 3]
```diff
@@ -17,16 +17,21 @@ compliant object storage, supporting SSO with SAML2 federated authentication.
 
 Project documentation is hosted on readthedocs: https://swift-browser-ui.rtfd.io
 
+Readmes on the additional APIs for container sharing, access requests, and
+better upload and download functionality are in their separate files.
+
+* Container sharing – `README-sharing.md`
```
Comment (blankdots) [inline README.md:23]: might as well link to them
```suggestion
* [Container sharing](README-sharing.md)
```

File: README.md [hunk 4]
```diff
@@ -17,16 +17,21 @@ compliant object storage, supporting SSO with SAML2 federated authentication.
...
+* Container access requests – `README-request.md`
```
Comment (blankdots) [inline README.md:24]:
```suggestion
* [Container access requests](README-request.md)
```

File: README.md [hunk 5]
```diff
@@ -17,16 +17,21 @@ compliant object storage, supporting SSO with SAML2 federated authentication.
...
+* Additional upload and download functionality `README-runner.md`
```
Comment (blankdots) [inline README.md:25]:
```suggestion
* [Additional upload and download functionality](README-runner.md)
```

File: README-runner.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+into object storage backends using Openstack Swift API. It will also make possible
+proper container uploads without plaintext passwords or tokens, to prevent
+token expiration.
+
+Additionally, the runner contains functionality for downloading whole containers
+as an archive without intermediary storage, as well as proxying downloads from
+shared containers, i.e. containers where the project has access via an ACL entry.
+
+### Installation
+`pip install git+https://github.com/cscfi/swift-upload-runner.git`
```
Comment (blankdots) [inline README-runner.md:13]: this might get deprecated

File: README-sharing.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+The backend contains a database for managing the information on which
+containers have been shared, by whom and to whom. The backend is meant
+to be used as a solution to centrally share the container access
+information in a given user pool, and thus isn't baked into the Openstack
+native command line tools.
+
+The default port for the service is 9090.
+
+### Usage
+```pip install . && swift-x-account-sharing```
```
Comment (blankdots) [inline README-sharing.md:21]: is this still valid

File: README-request.md (hunk 1 above)
Comment (sampsapenna) [reply README-request.md]: Yup, were forgotten, removed.

File: README-sharing.md (hunk 2 above)
Comment (sampsapenna) [reply README-sharing.md]: forgotten as well, removed

File: README.md (hunk 3 above)
Comment (sampsapenna) [reply README.md]: yup, moved over to relative links

File: README-sharing.md (hunk 7 above)
Comment (sampsapenna) [reply README-sharing.md:17]: this is valid, setup gives CLI entry points for all the subpackages with old syntax

File: README-runner.md (hunk 6 above)
Comment (sampsapenna) [reply README-runner.md]: yup will be depregated, changed over to similar install to the other readmes

File: docs/API_spec.yml [hunk 8]
```diff
@@ -0,0 +1,190 @@
+openapi: "3.0.2"
```
Comment (blankdots) [inline docs/API_spec.yml:1]: all the OpenAPIs files can be merged into one and grouped with tags https://swagger.io/docs/specification/grouping-operations-with-tags/

**Review by blankdots — APPROVED** [review]
the merge looks ok, I think we can improve in other issue the docker deployment to all use one docker file, but customise the entry point for each service

**Pull request merged**